### PR TITLE
[telemetry] add link metrics telemetry

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR adds telemetry uploading for Link Metrics telemetry.

### CODE REVIEW

File: src/utils/thread_helper.cpp [hunk 1]
```diff
@@ -1382,6 +1385,26 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher &aPublisher, threadn
...
+#if OTBR_ENABLE_LINK_METRICS_TELEMETRY
+    {
+        auto lowPowerMetrics = telemetryData.mutable_low_power_metrics();
+        // Begin of Link Metrics section.
+        for (const otNeighborInfo &neighborInfo : neighborTable)
+        {
+            otError             query_error;
+            otLinkMetricsValues values;
+
+            query_error = otLinkMetricsManagerGetMetricsValueByExtAddr(mInstance, &neighborInfo.mExtAddress, &values);
```
Comment (morningboata) [inline src/utils/thread_helper.cpp:1397]: Do you want to use VerifyOrExit here?

Comment (Irving-cl) [reply src/utils/thread_helper.cpp:1397]: Some neighbors doesn't support Link Metrics. So it is expected that some errors other than `OT_ERROR_NONE` happened.

Comment (morningboata) [reply src/utils/thread_helper.cpp:1397]: SG. Can you add a comment in the code about this?

Comment (Irving-cl) [reply src/utils/thread_helper.cpp:1397]: SG! Updated.

**Review by morningboata — APPROVED** [review]

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**